java-checks/src/main/java/org/sonar/java/checks/OverrideAnnotationCheck.java

```diff
@@ -50,12 +53,25 @@
   @Override
   public void visitNode(Tree tree) {
     MethodTreeImpl methodTree = (MethodTreeImpl) tree;
-    if(isOverriding(methodTree) && !methodTree.isAnnotatedOverride()){
+    if(isOverriding(methodTree) && !methodTree.isAnnotatedOverride() && !isExcluded(context.getJavaVersion(), methodTree)){
       addIssue(tree, "Add the \"@Override\" annotation above this method signature");
     }
   }
 
-  public boolean isOverriding(MethodTreeImpl methodTree) {
+  private static boolean isOverriding(MethodTreeImpl methodTree) {
     return BooleanUtils.isTrue(methodTree.isOverriding());
   }
+
+  private boolean isExcluded(JavaVersion javaVersion, MethodTreeImpl methodTree) {
+    if(javaVersion.isNotSet()) {
+      return false;
+    }
+    int javaIntVersion = javaVersion.asInt();
+    return javaIntVersion <= 4 || (javaIntVersion == 5 && (methodTree.symbol().owner().isInterface() || overrideFromInterface(methodTree))) ;
+  }
+
+  private static boolean overrideFromInterface(MethodTree methodTree) {
+    return ((JavaSymbol.MethodJavaSymbol) methodTree.symbol()).overriddenSymbol().owner().isInterface();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'overriddenSymbol' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)